### PR TITLE
Feature: add proptype eslint

### PR DESCRIPTION
**Pull request by KeshavChawla**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Code style update
- [ ] Refactor (refactoring or adding test which isn't a fix or add a feature)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes
- [ ] No

**Adviser warns of an error with netmask (high).**

Advisory code is 1658: https://www.npmjs.com/advisories/1658**

**Did you test your solution?**

- [x] I lightly tested it in one browser
- [ ] I deeply tested it in several browsers
- [ ] I wrote tests around it (unit tests, integration tests, E2E tests)

## Solution Description

Replaces the Jam3 checkProps for React PropTypes. 

## Side Effects, Risks, Impact

- [ ] N/A

**Additional comments:**

### CODE REVIEW

**Comment by namniak** [comment]
proptype eslint is different from the `@jam3/react-check-extra-props` because eslint checks whether all props that are used inside the component code are specified in the props. while `react-check-extra-props` checks prop types + warns if some properties that are not specified in the props are passed down from outside. 
also `eslint` isn't smart enough on it's own as it works on a file-by-file basis, meaning if props are defind outside of the component e.g. id `data/prop-types` file eslist will not reference it correctly and won't warn about any mismatch in props defined vs props used as it would do if prop types were defined in the same file. But this is a very common use case where we define props and re-use them and therefore put in separate data files.

The bottom line is `@jam3/react-check-extra-props` can't be really replaced with this linter as they serve different purposes. Maybe in the future, a combination of it may work if we get both to work together nicely but so far I think `@jam3/react-check-extra-props` brings more value over eslint based on my experience in using both in different projects.

@DonghyukJacobJang I guess this PR can be closed ad there's no progress here except removing `@jam3/react-check-extra-props`? and if in future we combine the two that would be a new PR that has it's own challenges

**Comment by edgardz** [comment]
@namniak @DonghyukJacobJang Related to https://github.com/Jam3/react-check-extra-props/pull/2

**Comment by namniak** [comment]
@edgardz thanks! will check it out. Yes it'd be ideal to be able to use both together 🎉

**Comment by neo** [comment]
I think once we have TypeScript, it would solve most of this problem for us without having to do any work :lolface:
